### PR TITLE
Hugo schmutz/DeFixmatch

### PR DESCRIPTION
**Pull request by HugoSchmutz**
## What does this PR do?

Fixes #\<116>

## Before submitting

- [ x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ x] Did you list all the **breaking changes** introduced by this pull request?

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ x] Check that all items from **Before submitting** are resolved
- [ x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Comment by HugoSchmutz** [comment]
@microsoft-github-policy-service agree

**Comment by HugoSchmutz** [comment]
> @HugoSchmutz please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by jindongwang** [comment]
@HugoSchmutz I think you may also add something new to the README.md file to refer to your algorithm.

**Comment by HugoSchmutz** [comment]
> @HugoSchmutz I think you may also add something new to the README.md file to refer to your algorithm.

Yes of course, in which part do you want me to add something?


I also did not update the results' csv as you may want to test the method by yourself.

**Comment by jindongwang** [comment]
If you are adding a whole new algorithm, you can just add news to the "news and updates" section to tell everyone that you added a new algorithm.

**Comment by Hhhhhhao** [comment]
> ## What does this PR do?
> Fixes #<116>
> 
> ### Does your PR introduce any breaking changes? If yes, please list them.
> ## Before submitting
> * [ x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
> * [ x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
> * [ x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
> * [ x] Did you make sure to **update the documentation** with your changes? (if necessary)
> * [ x] Did you list all the **breaking changes** introduced by this pull request?
> 
> ## PR review
> Anyone in the community is welcome to review the PR. Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:
> 
> * [ x] Is this pull request ready for review? (if not, please submit in draft mode)
> * [ x] Check that all items from **Before submitting** are resolved
> * [ x] Make sure the title is self-explanatory and the description concisely explains the PR
> * [ x] Add labels and milestones (and optionally projects) to the PR so it can be classified

Let me review this PR during the weekend (maybe just tomorrow). And we can then try to merge this PR into the main branch if it doesn't break anything. 

We also need to fix some minor bugs appeared in the recent issues for this version.

**Comment by HugoSchmutz** [comment]
> > ## What does this PR do?
> > Fixes #<116>
> > ### Does your PR introduce any breaking changes? If yes, please list them.
> > ## Before submitting
> > 
> > * [ x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
> > * [ x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
> > * [ x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
> > * [ x] Did you make sure to **update the documentation** with your changes? (if necessary)
> > * [ x] Did you list all the **breaking changes** introduced by this pull request?
> > 
> > ## PR review
> > Anyone in the community is welcome to review the PR. Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:
> > 
> > * [ x] Is this pull request ready for review? (if not, please submit in draft mode)
> > * [ x] Check that all items from **Before submitting** are resolved
> > * [ x] Make sure the title is self-explanatory and the description concisely explains the PR
> > * [ x] Add labels and milestones (and optionally projects) to the PR so it can be classified
> 
> Let me review this PR during the weekend (maybe just tomorrow). And we can then try to merge this PR into the main branch if it doesn't break anything.
> 
> We also need to fix some minor bugs appeared in the recent issues for this version.

Thank you a lot :)

**Comment by Hhhhhhao** [comment]
> > > ## What does this PR do?
> > > Fixes #<116>
> > > ### Does your PR introduce any breaking changes? If yes, please list them.
> > > ## Before submitting
> > > 
> > > * [ x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
> > > * [ x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
> > > * [ x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
> > > * [ x] Did you make sure to **update the documentation** with your changes? (if necessary)
> > > * [ x] Did you list all the **breaking changes** introduced by this pull request?
> > > 
> > > ## PR review
> > > Anyone in the community is welcome to review the PR. Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:
> > > 
> > > * [ x] Is this pull request ready for review? (if not, please submit in draft mode)
> > > * [ x] Check that all items from **Before submitting** are resolved
> > > * [ x] Make sure the title is self-explanatory and the description concisely explains the PR
> > > * [ x] Add labels and milestones (and optionally projects) to the PR so it can be classified
> > 
> > 
> > Let me review this PR during the weekend (maybe just tomorrow). And we can then try to merge this PR into the main branch if it doesn't break anything.
> > We also need to fix some minor bugs appeared in the recent issues for this version.
> 
> Thank you a lot :)

Hi Hugo, I reviewed the code and it looks good. As you mentioned you have run some experiments already using the code. Can you post those results and log files (in google drive or anyway more convenient for your) here?

File: README.md
Comment (Hhhhhhao) [inline README.md:1]: Maybe add results here?

**Review by Hhhhhhao — CHANGES_REQUESTED** [review]
Code looks good. Please post the results part too so we can add them for reference (for now).

File: README.md
Comment (HugoSchmutz) [reply README.md:1]: Good :)

Up to now, we ran all the USB's computer vision experiments but apparently, we deleted the logs. Here is the updated csv file. [usb_cv.csv](https://github.com/microsoft/Semi-supervised-learning/files/12067761/usb_cv.csv)


We are currently running the audio and NLP experiments but we are still missing several experiments. We have most of the log files.

- 1 run gtzan_100
- 1 run gtzan_400
- 2 runs fsdnoisy
- 3 runs urbansound and ESC
- 2 runs yahoo_answers_500
- 2 runs yahoo_answers_2000
- 2 runs yelp_review_250
- 2 runs yelp_review_1000

Unfortunately, we have not been able to preprocess Superbkbs using your script so we didn't run any experiments on it for now.

Comment (Hhhhhhao) [reply README.md:1]: Cool. Can you make an commit for this updated result file? After that we can merge. 
We will run the remaining and update the log files.

**Review by Hhhhhhao — APPROVED** [review]

**Pull request merged**
